src/H5FDsubfiling/H5FDsubfiling.c

```diff
@@ -564,6 +564,117 @@ H5Pget_fapl_subfiling(hid_t fapl_id, H5FD_subfiling_config_t *config_out)
+    H5F_t               *file_ptr      = NULL;
+    char               **filenames_arr = NULL;
+    char                *filepath      = NULL;
+    char                *subfile_dir   = NULL;
+    char                *base          = NULL;
+    int                  num_subfiles  = 0;
+    int                  num_digits    = 0;
+    herr_t               ret_value     = SUCCEED;
+
+    FUNC_ENTER_API(FAIL)
+
+    if (file_id < 0)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "invalid file ID");
+    if (!filenames)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "`filenames` was NULL");
+    if (!len)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "`len` was NULL");
+
+    *filenames = NULL;
+    *len       = 0;
+
+    if (NULL == (file_ptr = H5VL_object(file_id)))
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "invalid file ID");
+
+    if (H5FD_SUBFILING != H5F_get_driver_id(file_ptr))
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "file is not using Subfiling VFD");
+
+    if (H5F_shared_get_file_driver(H5F_SHARED(file_ptr), &driver) < 0)
+        HGOTO_ERROR(H5E_FILE, H5E_CANTGET, FAIL, "can't get driver structure from file ID");
+
+    if (NULL == (sf_context = H5FD__subfiling_get_object(((H5FD_subfiling_t *)driver)->context_id)))
+        HGOTO_ERROR(H5E_FILE, H5E_CANTGET, FAIL, "can't get subfiling context from ID");
+
+    if (!sf_context->topology)
+        HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL,
+                    "application topology hasn't been initialized yet for this file");
+
+    assert(sf_context->h5_file_id != UINT64_MAX);
+    assert(sf_context->h5_filename);
+    assert(sf_context->sf_num_subfiles > 0);
+    assert(sf_context->topology);
+
+    if (sf_context->topology->rank_is_ioc) {
+        assert(sf_context->sf_fids);
+        assert(sf_context->sf_num_fids > 0);
+
+        /* Get the basename of the full HDF5 filename */
+        if (H5_basename(sf_context->h5_filename, &base) < 0)
+            HGOTO_ERROR(H5E_RESOURCE, H5E_CANTALLOC, FAIL, "can't get HDF5 file basename");
+
+        /*
+         * Get the directory prefix where subfiles will be placed.
+         * Under normal circumstances, the subfiles are co-located
+         * with the HDF5 file, but users may specify a different
+         * directory name.
+         */
+        if (sf_context->subfile_prefix) {
+            if (NULL == (subfile_dir = H5MM_strdup(sf_context->subfile_prefix)))
+                HGOTO_ERROR(H5E_RESOURCE, H5E_CANTALLOC, FAIL, "couldn't copy subfile prefix");
+        }
+        else {
+            if (H5_dirname(sf_context->h5_filename, &subfile_dir) < 0)
+                HGOTO_ERROR(H5E_RESOURCE, H5E_CANTALLOC, FAIL, "couldn't get HDF5 file dirname");
+        }
+
+        num_subfiles = sf_context->sf_num_subfiles;
+        num_digits   = (int)(log10(num_subfiles) + 1);
+
+        if (NULL == (filenames_arr = calloc(1, (size_t)sf_context->sf_num_fids * sizeof(char *))))
```

Comment:
If you are callocing, it'd be more idiomatic to use ((size_t)sf_context->sf_num_fids, sizeof(char *)) instead of creating an array of size 1.